src/ruby.rs

```diff
@@ -69,7 +71,10 @@ pub fn entrypoint() -> Result<(), Exception> {
         io::stdin()
             .read_to_end(&mut program)
             .map_err(|_| IOError::new(&interp, "Could not read program from STDIN"))?;
-        let _ = interp.eval(program.as_slice())?;
```

Comment:
the `let _ = ...` here is saying: I don't want to deal with this error. Now we do want to deal with this error 🎉 

Instead of `map_err`, I think the code would communicate that intent more clearly by looking like the `-e` command handling below. Maybe something like:

```rust
if let Err(exc) = interp.eval(program.as_slice()) {
    handle_exception(&mut interp, &exc)?;
}
```

This code propagates failures to print stack traces the same way we propagate errors to read code from stdin.